services/database/user.ts

```diff
@@ -35,4 +48,163 @@ async function getUsers() {
     return users;
 }
 
-export { getUser, getUsers };
+/**
+ * Updates a user with the data corresponding to userInput
+ * @param userInput - data for the updated user
+ * @returns prisma User with updated information
+ */
+
+async function updateUser(userInput: UserInput): Promise<User> {
+    /*
+    Flow:
+    - get role and role data from UserInput
+    - create/update the role record
+    - update the user record
+    - return the user, including the role records
+    */
+    const roleData = userInput.role_data;
+    const user = await getUser(userInput.id);
+
+    switch (userInput.role) {
+        case roles.PARENT:
+            if (user.role && user.role !== roles.PARENT) {
```

Comment:
I'm not sure if checking `user.role` outside of the switch statement would make the code cleaner because if `user.role` is not null, then I need a switch statement to check whether the existing `user.role` is the same as the role being passed in, and then I would need another switch statement to actually call the appropriate function - correct me if I'm wrong